Makefile

```diff
@@ -1,10 +1,43 @@
-bin: deps
+CLASS_PATH = .
+CFLAGS = -O2 -fPIC
+DIR = p2pclient/java
+JAVA_INCLUDES = -I$(JAVA_HOME)/include/darwin -I$(JAVA_HOME)/include -I$(JAVA_HOME)/include/linux
+vpath %.class $(CLASS_PATH)
+
+.DEFAULT_GOAL := go-daemon
+
+java-daemon: libp2pd.jnilib
+
+libp2pd.jnilib : libp2pd.o
+	ld -dylib -flat_namespace -undefined suppress -macosx_version_min 10.13.4 -o $(DIR)/$@ $(DIR)/*.o -L$(DIR) -lp2pd
+
+libp2pd.o : libp2pd.a
+	gcc $(CFLAGS) -c $(DIR)/p2pd.c $(JAVA_INCLUDES) -o $(DIR)/$@
+
+libp2pd.a: p2pd.h
+	go build -o $(DIR)/$@ -buildmode=c-archive p2pd/main.go
+
+p2pd.h : p2pd.class
+	cd $(DIR) && javac -h $(CLASS_PATH) $*.java
+
+p2pd.class: go-daemon
```

Comment:
Will this be more efficient to avoid running `deps`?
```suggestion
p2pd.class: $(DIR)/p2pd.java
```

Reply:
> Will this be more efficient to avoid running deps?

@mhchia your suggestion will work if all the go source has been downloaded (e.g. github.com/libp2p/go-libp2p-connmgr), but if it is the first time you are pulling it down...I believe it will fail.  That being said, I understand the spirit of the suggestion and I will modify the build to only conditionally perform those steps.